typespec-extension/src/code-model-builder.ts

```diff
@@ -1570,53 +1574,17 @@ export class CodeModelBuilder {
       } else {
         // schema (usually JSON)
         let schema: Schema | undefined = undefined;
-        const verb = op.requests?.[0].protocol?.http?.method;
-        if (
-          // LRO, candidateResponseSchema is Model/ObjectSchema
-          candidateResponseSchema &&
-          candidateResponseSchema instanceof ObjectSchema &&
-          // bodyType is templated Model
-          bodyType.kind === "Model" &&
-          bodyType.templateMapper &&
-          bodyType.templateMapper.args &&
-          bodyType.templateMapper.args.length > 0
-        ) {
-          if (verb === "post") {
-            // for LRO ResourceAction, the standard does not require a final type, hence it can be the same as intermediate type
-            // https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#long-running-action-operations
-
-            // check if we can use candidateResponseSchema as response schema (instead of the templated Model), for LRO ResourceAction
-            if (candidateResponseSchema.properties?.length === bodyType.properties.size) {
-              let match = true;
-              for (const prop of Array.from(bodyType.properties.values())) {
-                const name = this.getName(prop);
-                if (!candidateResponseSchema.properties?.find((it) => it.language.default.name === name)) {
-                  match = false;
-                  break;
-                }
-              }
-              if (match) {
-                schema = candidateResponseSchema;
-                const sdkType = getClientType(this.sdkContext, bodyType);
-                const modelName = sdkType.kind === "model" ? sdkType.name : undefined;
-                this.trace(
-                  `Replace TypeSpec model '${modelName}' with '${candidateResponseSchema.language.default.name}'`,
-                );
-              }
-            }
-          } else if (verb === "delete") {
-            // for LRO ResourceDelete, final type will be replaced to "Void" in convenience API, hence do not generate the class
+        if (candidateResponseSchema) {
+          // The operation is LRO
+          if (trackConvenienceApi && !this.isArm()) {
+            // DPG uses the LroMetadata as poll/final result, not the response of activation request
+            // Therefore this schema will not be public.
             trackConvenienceApi = false;
           }
         }
```

Comment:
Not sure I understand the logic here. Why do we check whether `candidateResponseSchema`(which should be `lroMetadata.pollResultType`) exists before we set convenience usage for activation response type?

Reply:
After the removal of the code, `candidateResponseSchema` basically same as `isLro`

Reply:
Got it. Seems it can be simplified to `if (candidateResponseSchema && !this.isArm()){...}`.

Reply:
I've changed the signature to `longRunning: boolean`